mypy.ini

```diff
@@ -2,7 +2,8 @@
 follow_imports = skip
 check_untyped_defs = True
 disallow_untyped_defs = True
-files = tests/challenges/**/*.py
+exclude=tests/integration/test_image_gen.py  ;Because i was not able to figure out type annotation for now
+files = tests/challenges/**/*.py,tests/integration/memory/*.py ,tests/integration/agent_factory.py,tests/integration/test_execute_code.py,tests/integration/test_provider_openai.py,tests/integration/test_setup.py,tests/integration/test_web_selenium.py
```

Comment:
Why not `tests/**/*.py`?

Reply:
I have closed this pull request, because i am no longer working on this issue. I am currently working on another project which is more suited to my skills and interest,
Sorry for causing trouble.

Reply:
No worries, thanks anyways for taking an interest and putting the work in. :)